qemu/tests/virtio_console.py

```diff
@@ -1341,19 +1351,21 @@ def _virtio_dev_add(vm, pci_id, port_id, console="no"):
             port = "console-"
             port_type = "virtconsole"
         port += "%d-%d" % (pci_id, port_id)
-        ret = vm.monitors[0].cmd("device_add %s,"
-                                 "bus=virtio_serial_pci%d.0,"
-                                 "id=%s,"
-                                 "name=%s"
-                                 % (port_type, pci_id, port, port))
+        new_portdev = qdevices.QDevice(port_type)
+        for key, value in {'id': port, 'name': port, 'bus': "virtio_serial_pci"
+                           "%d.0" % pci_id}.items():
+            new_portdev.set_param(key, value)
+        (result, ver_out) = vm.devices.simple_hotplug(new_portdev, vm.monitor)
```

Comment:
simple_hotplug() is updated now, you will need give the parent bus, instead of `set_param('bus': 'virtio_serial_pci%d.0')` otherwise it will be plugged into pci.0:
```
15:55:06 ERROR| ERROR 1-Host_RHEL.m7.u5.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.5.x86_64.myprovider.virtio_console.spread_linear.virtserialport.hotplug.timeout_0 -> QMPCmdError: QMP command 'device_add' failed    (arguments: OrderedDict([('driver', 'virtserialport'), ('bus', 'pci.0'), ('id', 'serialport-1-0'), ('name', 'serialport-1-0')]),    error message: {u'class': u'GenericError', u'desc': u"Device 'virtserialport' can't go on PCI bus"})
```
There is no proper solutions since current `virtio-serial-pci` is only a `QDevice`, no Bus object related. Unless based on https://github.com/avocado-framework/avocado-vt/pull/1639/commits/eebdfa319c03f21bfe021dc5814fa3964222657f
So I guess we still need wait for it.

Reply:
@vivianQizhu Thanks, will update this patch when `eebdfa3` merged.